### PR TITLE
Feat: API of retrieving image registry secrets by image and its tag

### PR DESCRIPTION
**Pull request by zzxwill**
Add an API to retrieve image registry secrets by an image and or
its tag. In veluaUX, it will help auto-fill uesrs' imagePullSecrets
fields

Signed-off-by: Zheng Xi Zhou <zzxwill@gmail.com>

### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

http://127.0.0.1:8000/api/v1/projects/default/validate_image?image=nginx
{
 "data": {
  "existed": true,
  "secret": ""
 }
}

http://127.0.0.1:8000/api/v1/projects/default/validate_image?image=docker.io/nginx
{
 "data": {
  "existed": true,
  "secret": ""
 }
}

http://127.0.0.1:8000/api/v1/projects/default/validate_image?image=nginx3432423423
{
 "BusinessCode": 500,
 "Message": "GET https://index.docker.io/v2/library/nginx3432423423/manifests/latest: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:library/nginx3432423423 Type:repository]]"
}

http://127.0.0.1:8000/api/v1/projects/default/validate_image?image=registry.cn-beijing.aliyuncs.com/vela/nginx:latest
{
 "data": {
  "existed": true,
  "secret": "acr"
 }
}

### CODE REVIEW

**Review by wonderflow — APPROVED** [review]
good job

**Review by barnettZQG — CHANGES_REQUESTED** [review]
There are some problems that do not meet the requirements:

1. We need an API to check the existence  and query the image base info, including:
   * container ports
   * volumes
   * registry name and secret name
   * (option)image sizes and update time
2. This API should support all public registries and configured private registries.

**Comment by zzxwill** [comment]
> There are some problems that do not meet the requirements:
> 
> 1. We need an API to check the existence  and query the image base info, including:
>    
>    * container ports
>    * volumes
>    * registry name and secret name
>    * (option)image sizes and update time
> 
> > Advised API path: /api/v1/repository/oci/config?image=xxx
> 
> 2. This API should support all public registries and configured private registries.

This PR is to deliver feature https://github.com/kubevela/velaux/issues/436.

File: pkg/apiserver/rest/usecase/project.go [hunk 1]
```diff
@@ -613,3 +617,54 @@ func retrieveConfigFromApplication(a v1beta1.Application, project string) *apisv
 		Description:       a.Annotations[types.AnnotationConfigDescription],
 	}
 }
+
+func (p *projectUsecaseImpl) ValidateImage(ctx context.Context, projectName, image string) (*apisv1.ImageResponse, error) {
+	return validateImage(ctx, p.k8sClient, projectName, image)
+}
+
+func validateImage(ctx context.Context, k8sClient client.Client, project, imageName string) (*apisv1.ImageResponse, error) {
```
Comment (barnettZQG) [inline pkg/apiserver/rest/usecase/project.go:625]: Please add test cases for private repositories, including the docker hub.

File: pkg/apiserver/rest/usecase/project.go [hunk 2]
```diff
@@ -613,3 +617,54 @@ func retrieveConfigFromApplication(a v1beta1.Application, project string) *apisv
...
+
+	for _, s := range secrets.Items {
+		if s.Labels[types.LabelConfigProject] == "" || s.Labels[types.LabelConfigProject] == project {
+			conf := s.Data[".dockerconfigjson"]
+			var auths map[string]map[string]map[string][]byte
+			if err := json.Unmarshal(conf, &auths); err != nil {
+				return nil, err
+			}
+			imagePullSecret = s.Name
+			data := auths["auths"][imageURL]["auth"]
```
Comment (barnettZQG) [inline pkg/apiserver/rest/usecase/project.go:651]: There is a nil-pointer risk.

Comment (zzxwill) [reply pkg/apiserver/rest/usecase/project.go]: Fixed, thanks.

File: pkg/apiserver/rest/usecase/project.go (hunk 1 above)
Comment (zzxwill) [reply pkg/apiserver/rest/usecase/project.go:626]: Done, thanks.

**Review by barnettZQG — APPROVED** [review]
LGTM, there is a new permission resource with the name `project/image`, we need to change the default policy.

**Pull request merged**